packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -26,6 +26,11 @@ exports.onPostBuild = (args, pluginOptions) => {
       `${rootDir}/offline-plugin-app-shell-fallback/index.html`,
     ],
     stripPrefix: rootDir,
+    // If `pathPrefix` is configured by user, we should replace
+    // the `public` prefix with `pathPrefix`.
+    // See more at:
+    // https://github.com/GoogleChrome/sw-precache#replaceprefix-string
+    replacePrefix: args.pathPrefix || '',
```

Comment:
You also need to check if the user is actually building with paths prefixed. You can set a pathPrefix but build without them e.g. for local testing.

https://github.com/gatsbyjs/gatsby/blob/cd68468c0f02dfb1700b464b91da518aa3dd3884/packages/gatsby/src/utils/webpack.config.js#L88-L89

Reply:
@KyleAMathews I think the `pathPrefix` will only be set if `program.prefixPaths === true` for plugin API(`onPostBuild` here), otherwise it will be `''`.

https://github.com/gatsbyjs/gatsby/blob/3f5c54d1bff877d8b28254aeae2221554f330ef7/packages/gatsby/src/utils/api-runner-node.js#L53-L55

> P.S.  
> It seems no full documents for plugin API like `onPostBuild` now, I got the `pathPrefix` argument by referring the source 😅. And can I get the `program.prefixPaths` variable in `onPostBuild` function?

Reply:
Haha so it is :-)

Reply:
Forgot I did that… :-)